txjuju/cli.py

```diff
@@ -13,9 +14,124 @@
 from twisted.internet.protocol import ProcessProtocol
 from twisted.python import log
 
+from . import config, _utils
 from .errors import CLIError
 
 
+def get_executable(filename, version_cli, cfgdir, envvars=None):
+    """Return the Executable for the given juju binary.
+
+    @param filename: The path to the juju binary.
+    @param version_cli: The version-specific JujuXCLI to use.
+    @param cfgdir: The Juju config dir to use.
+    @param envvars: The additional environment variables to use.
+    """
+    if not version_cli:
```

Comment:
This is just a safety net in case someone accidentally passes in None (or empty string, etc.).  In practice it doesn't add much, but it doesn't cost much to do now and if the error happens just once it has more than paid for itself.